src/multiCursor.ts

```diff
@@ -0,0 +1,19 @@
+import { Command, EditorView } from '@codemirror/view';
+import { EditorSelection } from '@codemirror/state';
+
+const addCursorFactory =
+  (direction: 'up' | 'down'): Command =>
+  (view) => {
+    const forward = direction === 'down';
+    const current = view.state.selection.ranges;
+    const added = current.map((r) => view.moveVertically(r, forward));
+
+    view.dispatch({
+      selection: EditorSelection.create([...added, ...current], 0),
```

Comment:
FYI you can use `addRange` here: https://codemirror.net/6/docs/ref/#state.EditorSelection.addRange

prolly better to stick w CM built-ins for these sort of ops

Reply:
There's many ways to write this, I don't think using `addRange` is more correct or built-in than this

Reply:
I love to please my reviewers, so there you go, `addRange` is in full force!